UI/CUIControls.cpp

```diff
@@ -1924,3 +1928,139 @@ void RotatingGraphic::Render() {
     glPopClientAttrib();
     glPopMatrix();
 }
+
+////////////////////////////////////////////////
+// CUIToggleButton
+////////////////////////////////////////////////
+CUIToggleRepresenter::CUIToggleRepresenter(const std::string& tooltip,
+                                           boost::shared_ptr<GG::SubTexture> icon,
+                                           const GG::Clr& highlight_clr) :
+                                           m_unchecked_icon(icon),
+                                           m_checked_icon(icon),
+                                           m_unchecked_color(highlight_clr),
+                                           m_checked_color(highlight_clr),
+                                           m_tooltip(tooltip)
+{}
+
+boost::shared_ptr<GG::SubTexture> CUIToggleRepresenter::GetUncheckedIcon() const
+{ return m_unchecked_icon; }
+
+boost::shared_ptr<GG::SubTexture> CUIToggleRepresenter::GetCheckedIcon() const
+{ return m_checked_icon; }
+
+const bool CUIToggleRepresenter::HasSharedIcon() const
+{ return (m_checked_icon == m_unchecked_icon); }
+
+const GG::Clr& CUIToggleRepresenter::GetUncheckedColor() const
+{ return m_unchecked_color; }
+
+const GG::Clr& CUIToggleRepresenter::GetCheckedColor() const
+{ return m_checked_color; }
+
+void CUIToggleRepresenter::SetTooltip(const std::string& tooltip)
+{ m_tooltip = tooltip; }
+
+/** \brief Set the SubTexture for both checked and unchecked states
+ * 
+ * @param[in] icon SubTexture to assign
+ */
+void CUIToggleRepresenter::SetIcon(boost::shared_ptr<GG::SubTexture> icon) {
```

Comment:
Put the comments in the header...

Why do you need this function, when the next two together do the same thing?
Why do you need these three functions, when constructor parameters could be used to set the icons?